File: Admin/admin.client/src/api/rq/document/DocumentQueryRQ.ts
import { SystemDocumentListRQ } from "@etsoo/smarterp-core";

/**
 * Document query request data
 * 文档查询请求数据
 */
export type DocumentQueryRQ = SystemDocumentListRQ & {
  /**
   * Organizaton id
   * 机构编号
   */
  orgId?: number;

  /**
   * System template or not
   * 系统模板与否
   */
  systemTemplate?: boolean;

  /**
   * Has parameters or not
   * 是否有参数
   */
  hasParameters?: boolean;
};

File: Admin/Admin.Server/RQ/Document/DocumentQueryRQ.cs
using PlatformShared.RQ;

namespace Admin.Server.RQ.Document
{
    /// <summary>
    /// Document query request data
    /// 文档查询请求数据
    /// </summary>
    public record DocumentQueryRQ : SystemDocumentListRQ
    {
        /// <summary>
        /// Organizaton id
        /// 机构编号
        /// </summary>
        public int? OrgId { get; init; }

        /// <summary>
        /// Is system template or not
        /// 是否是系统模板
        /// </summary>
        public bool? SystemTemplate { get; init; }

        /// <summary>
        /// Has parameters or not
        /// 是否有参数
        /// </summary>
        public bool? HasParameters { get; init; }
    }
}
